receiver/prometheusreceiver/internal/vm_histograms.go

```diff
@@ -0,0 +1,84 @@
+package internal
+
+import (
+	"errors"
+	"math"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+// This file contains logic for handling VictoriaMetrics Histograms.
+// See: https://docs.victoriametrics.com/victoriametrics/keyconcepts/#histogram
+
+const (
+	vmHistogramRangeLabel = "vmrange"
+)
+
+var (
+	errInvalidVMHistogramRange = errors.New("invalid 'vmrange' label value on histogram bucket")
+)
+
+func vmHistogramParseRange(vmrange string) (start, end float64, err error) {
+	before, after, ok := strings.Cut(vmrange, "...")
+	if !ok {
+		return 0, 0, errInvalidVMHistogramRange
+	}
+	start, err = strconv.ParseFloat(before, 64)
+	if err != nil {
+		return 0, 0, errInvalidVMHistogramRange
+	}
+	end, err = strconv.ParseFloat(after, 64)
+	if err != nil {
+		return 0, 0, errInvalidVMHistogramRange
+	}
+	return start, end, nil
+}
+
+func vmHistogramGetScale(start, end float64) int32 {
+	// Compute "scale" as per the OTLP spec:
+	//   https://opentelemetry.io/docs/specs/otel/metrics/data-model/#exponential-scale
+	//
+	//  base = 2**(2**(-scale)) = end / start
+	return int32(math.Round(-math.Log2(math.Log2(end / start))))
+}
+
+func vmConvertBuckets(dest pmetric.ExponentialHistogramDataPoint, source []*dataPoint) {
+	// base is the multiplier from one bucket boundary to the next:
+	//   base = 2**(2**(-scale))
+	// To calculate a logarithm with a base of "base", we'll use the formula:
+	//   log_base(x) = log2(x) / log2(base)
+	// So what we actually want to know is log2(base) = 2**(-scale)
+	log2base := math.Pow(2, float64(-dest.Scale()))
```

Comment:
worked through the math myself, this makes sense to me! 
i.e
````
// vmLe = converted upper le bound from base X with some rounding error
[1] base = 2**(2**(-scale))
[2] base ** index < vmLe <= base**(index+1)
[3] vmLeIndex = toIntegerFunc(log_base(vmLe)) 
[4] index < vmLeIndex <= index + 1
```
Food for thought (need to think the math over, but the friday bug is getting to me)...
If `toIntegerFunc = math.Round` like in L66, wouldn't we have the change of having `index == vmLeIndex` which adds another layer of error to the conversion.
What about using `toIntegerFunc = math.Ceiling` instead?
Such that if we apply `math.Ceiling` to [4], it'd become `math.Ceiling(vmLeIndex) == index + 1`